### ISSUE TITLE
ICE when compiling with -O1 -mllvm -opt-bisect-limit=10 -g

### ISSUE THREAD

**Issue by theo-lw** [llvm/llvm-project#53937]
This occurs in `preprocessed.c` below:

```c
static short a = 5;
char b;
int c, d, e = 2, g;
void h() {
  a = 0;
  for (; b;)
    a = c;
}
int main() {
  e = ({
    unsigned f;
  as:
    g = f = ~(-a / e ^ b ^ ~f << 7);
    if (d) {
      printf("l_311=%llu\n", (long long)5);
      goto as;
    }
    a;
  });
}
```

```
$ clang-trunk -v
clang version 15.0.0 (https://github.com/llvm/llvm-project.git 06f346cb7238fff88b2c7f94bf55fde50de5c1dd)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/cnsun/usr/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/10
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/8
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/9
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/10
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Candidate multilib: x32;@mx32
Selected multilib: .;@m64
$
$ clang-trunk -w -O1 -mllvm -opt-bisect-limit=10 -g preprocessed.c
BISECT: running pass (1) Annotation2MetadataPass on [module]
BISECT: running pass (2) ForceFunctionAttrsPass on [module]
BISECT: running pass (3) InferFunctionAttrsPass on [module]
BISECT: running pass (4) LowerExpectIntrinsicPass on h
BISECT: running pass (5) SimplifyCFGPass on h
BISECT: running pass (6) SROAPass on h
BISECT: running pass (7) EarlyCSEPass on h
BISECT: running pass (8) LowerExpectIntrinsicPass on main
BISECT: running pass (9) SimplifyCFGPass on main
BISECT: running pass (10) SROAPass on main
BISECT: NOT running pass (11) EarlyCSEPass on main
BISECT: NOT running pass (12) OpenMPOptPass on [module]
BISECT: NOT running pass (13) IPSCCPPass on [module]
BISECT: NOT running pass (14) CalledValuePropagationPass on [module]
BISECT: NOT running pass (15) GlobalOptPass on [module]
BISECT: NOT running pass (16) PromotePass on h
BISECT: NOT running pass (17) PromotePass on main
BISECT: NOT running pass (18) DeadArgumentEliminationPass on [module]
BISECT: NOT running pass (19) InstCombinePass on h
BISECT: NOT running pass (20) SimplifyCFGPass on h
BISECT: NOT running pass (21) InstCombinePass on main
BISECT: NOT running pass (22) SimplifyCFGPass on main
BISECT: NOT running pass (23) InvalidateAnalysisPass<llvm::AAManager> on h
BISECT: NOT running pass (24) InvalidateAnalysisPass<llvm::AAManager> on main
BISECT: NOT running pass (25) InlinerPass on (h)
BISECT: NOT running pass (26) InlinerPass on (h)
BISECT: NOT running pass (27) PostOrderFunctionAttrsPass on (h)
BISECT: NOT running pass (28) SROAPass on h
BISECT: NOT running pass (29) EarlyCSEPass on h
BISECT: NOT running pass (30) SimplifyCFGPass on h
BISECT: NOT running pass (31) InstCombinePass on h
BISECT: NOT running pass (32) LibCallsShrinkWrapPass on h
BISECT: NOT running pass (33) SimplifyCFGPass on h
BISECT: NOT running pass (34) ReassociatePass on h
BISECT: NOT running pass (35) LoopSimplifyPass on h
BISECT: NOT running pass (36) LCSSAPass on h
BISECT: NOT running pass (37) LoopInstSimplifyPass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (38) LoopSimplifyCFGPass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (39) LICMPass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (40) LoopRotatePass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (41) LICMPass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (42) SimpleLoopUnswitchPass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (43) SimplifyCFGPass on h
BISECT: NOT running pass (44) InstCombinePass on h
BISECT: NOT running pass (45) LoopSimplifyPass on h
BISECT: NOT running pass (46) LCSSAPass on h
BISECT: NOT running pass (47) LoopIdiomRecognizePass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (48) IndVarSimplifyPass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (49) LoopDeletionPass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (50) LoopFullUnrollPass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (51) SROAPass on h
BISECT: NOT running pass (52) MemCpyOptPass on h
BISECT: NOT running pass (53) SCCPPass on h
BISECT: NOT running pass (54) BDCEPass on h
BISECT: NOT running pass (55) InstCombinePass on h
BISECT: NOT running pass (56) ADCEPass on h
BISECT: NOT running pass (57) SimplifyCFGPass on h
BISECT: NOT running pass (58) InstCombinePass on h
BISECT: NOT running pass (59) InlinerPass on (main)
BISECT: NOT running pass (60) InlinerPass on (main)
BISECT: NOT running pass (61) PostOrderFunctionAttrsPass on (main)
BISECT: NOT running pass (62) SROAPass on main
BISECT: NOT running pass (63) EarlyCSEPass on main
BISECT: NOT running pass (64) SimplifyCFGPass on main
BISECT: NOT running pass (65) InstCombinePass on main
BISECT: NOT running pass (66) LibCallsShrinkWrapPass on main
BISECT: NOT running pass (67) SimplifyCFGPass on main
BISECT: NOT running pass (68) ReassociatePass on main
BISECT: NOT running pass (69) LoopSimplifyPass on main
BISECT: NOT running pass (70) LCSSAPass on main
BISECT: NOT running pass (71) LoopInstSimplifyPass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (72) LoopSimplifyCFGPass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (73) LICMPass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (74) LoopRotatePass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (75) LICMPass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (76) SimpleLoopUnswitchPass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (77) SimplifyCFGPass on main
BISECT: NOT running pass (78) InstCombinePass on main
BISECT: NOT running pass (79) LoopSimplifyPass on main
BISECT: NOT running pass (80) LCSSAPass on main
BISECT: NOT running pass (81) LoopIdiomRecognizePass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (82) IndVarSimplifyPass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (83) LoopDeletionPass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (84) LoopFullUnrollPass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (85) SROAPass on main
BISECT: NOT running pass (86) MemCpyOptPass on main
BISECT: NOT running pass (87) SCCPPass on main
BISECT: NOT running pass (88) BDCEPass on main
BISECT: NOT running pass (89) InstCombinePass on main
BISECT: NOT running pass (90) ADCEPass on main
BISECT: NOT running pass (91) SimplifyCFGPass on main
BISECT: NOT running pass (92) InstCombinePass on main
BISECT: NOT running pass (93) GlobalOptPass on [module]
BISECT: NOT running pass (94) GlobalDCEPass on [module]
BISECT: NOT running pass (95) EliminateAvailableExternallyPass on [module]
BISECT: NOT running pass (96) ReversePostOrderFunctionAttrsPass on [module]
BISECT: NOT running pass (97) Float2IntPass on h
BISECT: NOT running pass (98) LowerConstantIntrinsicsPass on h
BISECT: NOT running pass (99) LoopSimplifyPass on h
BISECT: NOT running pass (100) LCSSAPass on h
BISECT: NOT running pass (101) LoopRotatePass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (102) LoopDeletionPass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (103) LoopDistributePass on h
BISECT: NOT running pass (104) InjectTLIMappings on h
BISECT: NOT running pass (105) LoopVectorizePass on h
BISECT: NOT running pass (106) LoopLoadEliminationPass on h
BISECT: NOT running pass (107) InstCombinePass on h
BISECT: NOT running pass (108) SimplifyCFGPass on h
BISECT: NOT running pass (109) VectorCombinePass on h
BISECT: NOT running pass (110) InstCombinePass on h
BISECT: NOT running pass (111) LoopUnrollPass on h
BISECT: NOT running pass (112) WarnMissedTransformationsPass on h
BISECT: NOT running pass (113) InstCombinePass on h
BISECT: NOT running pass (114) LoopSimplifyPass on h
BISECT: NOT running pass (115) LCSSAPass on h
BISECT: NOT running pass (116) LICMPass on Loop at depth 1 containing: %for.cond<header><exiting>,%for.body<latch>
BISECT: NOT running pass (117) AlignmentFromAssumptionsPass on h
BISECT: NOT running pass (118) LoopSinkPass on h
BISECT: NOT running pass (119) InstSimplifyPass on h
BISECT: NOT running pass (120) DivRemPairsPass on h
BISECT: NOT running pass (121) SimplifyCFGPass on h
BISECT: NOT running pass (122) Float2IntPass on main
BISECT: NOT running pass (123) LowerConstantIntrinsicsPass on main
BISECT: NOT running pass (124) LoopSimplifyPass on main
BISECT: NOT running pass (125) LCSSAPass on main
BISECT: NOT running pass (126) LoopRotatePass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (127) LoopDeletionPass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (128) LoopDistributePass on main
BISECT: NOT running pass (129) InjectTLIMappings on main
BISECT: NOT running pass (130) LoopVectorizePass on main
BISECT: NOT running pass (131) LoopLoadEliminationPass on main
BISECT: NOT running pass (132) InstCombinePass on main
BISECT: NOT running pass (133) SimplifyCFGPass on main
BISECT: NOT running pass (134) VectorCombinePass on main
BISECT: NOT running pass (135) InstCombinePass on main
BISECT: NOT running pass (136) LoopUnrollPass on main
BISECT: NOT running pass (137) WarnMissedTransformationsPass on main
BISECT: NOT running pass (138) InstCombinePass on main
BISECT: NOT running pass (139) LoopSimplifyPass on main
BISECT: NOT running pass (140) LCSSAPass on main
BISECT: NOT running pass (141) LICMPass on Loop at depth 1 containing: %as<header><exiting>,%if.then<latch>
BISECT: NOT running pass (142) AlignmentFromAssumptionsPass on main
BISECT: NOT running pass (143) LoopSinkPass on main
BISECT: NOT running pass (144) InstSimplifyPass on main
BISECT: NOT running pass (145) DivRemPairsPass on main
BISECT: NOT running pass (146) SimplifyCFGPass on main
BISECT: NOT running pass (147) CGProfilePass on [module]
BISECT: NOT running pass (148) GlobalDCEPass on [module]
BISECT: NOT running pass (149) ConstantMergePass on [module]
BISECT: NOT running pass (150) RelLookupTableConverterPass on [module]
BISECT: NOT running pass (151) Canonicalize Freeze Instructions in Loops on loop
BISECT: NOT running pass (152) Loop Strength Reduction on loop
BISECT: NOT running pass (153) Merge contiguous icmps into a memcmp on function (h)
BISECT: NOT running pass (154) Expand memcmp() to load/stores on function (h)
BISECT: NOT running pass (155) Constant Hoisting on function (h)
BISECT: NOT running pass (156) Partially inline calls to library functions on function (h)
BISECT: NOT running pass (157) X86 Partial Reduction on function (h)
BISECT: NOT running pass (158) CodeGen Prepare on function (h)
BISECT: NOT running pass (159) X86 DAG->DAG Instruction Selection on function (h)
BISECT: NOT running pass (160) Local Dynamic TLS Access Clean-up on function (h)
BISECT: NOT running pass (161) X86 Domain Reassignment Pass on function (h)
BISECT: NOT running pass (162) Early Tail Duplication on function (h)
BISECT: NOT running pass (163) Optimize machine instruction PHIs on function (h)
BISECT: NOT running pass (164) Remove dead machine instructions on function (h)
BISECT: NOT running pass (165) Early If-Conversion on function (h)
BISECT: NOT running pass (166) X86 cmov Conversion on function (h)
BISECT: NOT running pass (167) Early Machine Loop Invariant Code Motion on function (h)
BISECT: NOT running pass (168) Machine Common Subexpression Elimination on function (h)
BISECT: NOT running pass (169) Machine code sinking on function (h)
BISECT: NOT running pass (170) Peephole Optimizations on function (h)
BISECT: NOT running pass (171) Remove dead machine instructions on function (h)
BISECT: NOT running pass (172) Live Range Shrink on function (h)
BISECT: NOT running pass (173) X86 LEA Optimize on function (h)
BISECT: NOT running pass (174) X86 Optimize Call Frame on function (h)
BISECT: NOT running pass (175) X86 Avoid Store Forwarding Blocks on function (h)
BISECT: NOT running pass (176) Two-Address instruction pass on function (h)
BISECT: NOT running pass (177) Machine Instruction Scheduler on function (h)
BISECT: NOT running pass (178) Stack Slot Coloring on function (h)
BISECT: NOT running pass (179) Machine Copy Propagation Pass on function (h)
BISECT: NOT running pass (180) Machine Loop Invariant Code Motion on function (h)
BISECT: NOT running pass (181) Fixup Statepoint Caller Saved on function (h)
BISECT: NOT running pass (182) PostRA Machine Sink on function (h)
BISECT: NOT running pass (183) Shrink Wrapping analysis on function (h)
BISECT: NOT running pass (184) Control Flow Optimizer on function (h)
BISECT: NOT running pass (185) Tail Duplication on function (h)
BISECT: NOT running pass (186) Machine Copy Propagation Pass on function (h)
BISECT: NOT running pass (187) Post RA top-down list latency scheduler on function (h)
BISECT: NOT running pass (188) Branch Probability Basic Block Placement on function (h)
BISECT: NOT running pass (189) X86 Execution Dependency Fix on function (h)
BISECT: NOT running pass (190) BreakFalseDeps on function (h)
BISECT: NOT running pass (191) X86 Byte/Word Instruction Fixup on function (h)
BISECT: NOT running pass (192) X86 Atom pad short functions on function (h)
BISECT: NOT running pass (193) X86 LEA Fixup on function (h)
BISECT: NOT running pass (194) Canonicalize Freeze Instructions in Loops on loop
BISECT: NOT running pass (195) Loop Strength Reduction on loop
BISECT: NOT running pass (196) Merge contiguous icmps into a memcmp on function (main)
BISECT: NOT running pass (197) Expand memcmp() to load/stores on function (main)
BISECT: NOT running pass (198) Constant Hoisting on function (main)
BISECT: NOT running pass (199) Partially inline calls to library functions on function (main)
BISECT: NOT running pass (200) X86 Partial Reduction on function (main)
BISECT: NOT running pass (201) CodeGen Prepare on function (main)
BISECT: NOT running pass (202) X86 DAG->DAG Instruction Selection on function (main)
BISECT: NOT running pass (203) Local Dynamic TLS Access Clean-up on function (main)
BISECT: NOT running pass (204) X86 Domain Reassignment Pass on function (main)
BISECT: NOT running pass (205) Early Tail Duplication on function (main)
BISECT: NOT running pass (206) Optimize machine instruction PHIs on function (main)
BISECT: NOT running pass (207) Remove dead machine instructions on function (main)
BISECT: NOT running pass (208) Early If-Conversion on function (main)
BISECT: NOT running pass (209) X86 cmov Conversion on function (main)
BISECT: NOT running pass (210) Early Machine Loop Invariant Code Motion on function (main)
BISECT: NOT running pass (211) Machine Common Subexpression Elimination on function (main)
BISECT: NOT running pass (212) Machine code sinking on function (main)
BISECT: NOT running pass (213) Peephole Optimizations on function (main)
BISECT: NOT running pass (214) Remove dead machine instructions on function (main)
BISECT: NOT running pass (215) Live Range Shrink on function (main)
BISECT: NOT running pass (216) X86 LEA Optimize on function (main)
BISECT: NOT running pass (217) X86 Optimize Call Frame on function (main)
BISECT: NOT running pass (218) X86 Avoid Store Forwarding Blocks on function (main)
BISECT: NOT running pass (219) Two-Address instruction pass on function (main)
BISECT: NOT running pass (220) Machine Instruction Scheduler on function (main)
clang-15: /tmp/tmp.UowgjXRJkU-clang-builder/llvm-project/llvm/lib/CodeGen/LiveDebugVariables.cpp:886: llvm::MachineBasicBlock::iterator {anonymous}::LDVImpl::handleDebugInstr(llvm::MachineInstr&, llvm::SlotIndex): Assertion `!MI.getOperand(0).isReg() || !MI.getOperand(0).getReg().isVirtual()' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: /scratch/software/clang-trunk/bin/clang-15 -cc1 -triple x86_64-unknown-linux-gnu -emit-obj --mrelax-relocations -disable-free -clear-ast-before-backend -main-file-name preprocessed.c -mrelocation-model static -mframe-pointer=none -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -mllvm -treat-scalable-fixed-error-as-warning -debug-info-kind=constructor -dwarf-version=5 -debugger-tuning=gdb -fcoverage-compilation-dir=/home/tlwang/clang_test/_scratch_cnsun_workspace_compiler-testing-infrastructure_ccmd_compilable-programs_t.c-1191419-1643383090-m32-12332.c -resource-dir /scratch/software/clang-trunk/lib/clang/15.0.0 -internal-isystem /scratch/software/clang-trunk/lib/clang/15.0.0/include -internal-isystem /usr/local/include -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/10/../../../../x86_64-linux-gnu/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem /include -internal-externc-isystem /usr/include -O1 -w -fdebug-compilation-dir=/home/tlwang/clang_test/_scratch_cnsun_workspace_compiler-testing-infrastructure_ccmd_compilable-programs_t.c-1191419-1643383090-m32-12332.c -ferror-limit 19 -fgnuc-version=4.2.1 -fcolor-diagnostics -mllvm -opt-bisect-limit=10 -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /tmp/preprocessed-2dbacf.o -x c preprocessed.c
1.	<eof> parser at end of file
2.	Code generation
3.	Running pass 'Function Pass Manager' on module 'preprocessed.c'.
4.	Running pass 'Debug Variable Analysis' on function '@main'
 #0 0x00005579e28853b4 PrintStackTraceSignalHandler(void*) Signals.cpp:0:0
 #1 0x00005579e2882ade SignalHandler(int) Signals.cpp:0:0
 #2 0x00007f0e9e4e43c0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x153c0)
 #3 0x00007f0e9df8318b raise /build/glibc-eX1tMB/glibc-2.31/signal/../sysdeps/unix/sysv/linux/raise.c:51:1
 #4 0x00007f0e9df62859 abort /build/glibc-eX1tMB/glibc-2.31/stdlib/abort.c:81:7
 #5 0x00007f0e9df62729 get_sysdep_segment_value /build/glibc-eX1tMB/glibc-2.31/intl/loadmsgcat.c:509:8
 #6 0x00007f0e9df62729 _nl_load_domain /build/glibc-eX1tMB/glibc-2.31/intl/loadmsgcat.c:970:34
 #7 0x00007f0e9df73f36 (/lib/x86_64-linux-gnu/libc.so.6+0x36f36)
 #8 0x00005579e1d15ae3 llvm::LiveDebugVariables::runOnMachineFunction(llvm::MachineFunction&) (/scratch/software/clang-trunk/bin/clang-15+0x3115ae3)
 #9 0x00005579e19a7948 llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (/scratch/software/clang-trunk/bin/clang-15+0x2da7948)
#10 0x00005579e1eccc75 llvm::FPPassManager::runOnFunction(llvm::Function&) (/scratch/software/clang-trunk/bin/clang-15+0x32ccc75)
#11 0x00005579e1ecceb9 llvm::FPPassManager::runOnModule(llvm::Module&) (/scratch/software/clang-trunk/bin/clang-15+0x32cceb9)
#12 0x00005579e1ecd702 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/scratch/software/clang-trunk/bin/clang-15+0x32cd702)
#13 0x00005579e2c084b7 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) (/scratch/software/clang-trunk/bin/clang-15+0x40084b7)
#14 0x00005579e3a90923 clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (/scratch/software/clang-trunk/bin/clang-15+0x4e90923)
#15 0x00005579e4b7a629 clang::ParseAST(clang::Sema&, bool, bool) (/scratch/software/clang-trunk/bin/clang-15+0x5f7a629)
#16 0x00005579e3a8f298 clang::CodeGenAction::ExecuteAction() (/scratch/software/clang-trunk/bin/clang-15+0x4e8f298)
#17 0x00005579e336bcc9 clang::FrontendAction::Execute() (/scratch/software/clang-trunk/bin/clang-15+0x476bcc9)
#18 0x00005579e32f854e clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/scratch/software/clang-trunk/bin/clang-15+0x46f854e)
#19 0x00005579e34483d3 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/scratch/software/clang-trunk/bin/clang-15+0x48483d3)
#20 0x00005579dfee3a01 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/scratch/software/clang-trunk/bin/clang-15+0x12e3a01)
#21 0x00005579dfedfdd8 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&) driver.cpp:0:0
#22 0x00005579dfe02fd9 main (/scratch/software/clang-trunk/bin/clang-15+0x1202fd9)
#23 0x00007f0e9df640b3 __libc_start_main /build/glibc-eX1tMB/glibc-2.31/csu/../csu/libc-start.c:342:3
#24 0x00005579dfedf94e _start (/scratch/software/clang-trunk/bin/clang-15+0x12df94e)
clang-15: error: unable to execute command: Aborted (core dumped)
clang-15: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 15.0.0 (https://github.com/llvm/llvm-project.git 06f346cb7238fff88b2c7f94bf55fde50de5c1dd)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/cnsun/usr/bin
clang-15: note: diagnostic msg:
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-15: note: diagnostic msg: /tmp/preprocessed-85ecee.c
clang-15: note: diagnostic msg: /tmp/preprocessed-85ecee.sh
clang-15: note: diagnostic msg:

********************
```

**Comment by sylvain-audi** [comment]
I encountered a crash that seems identical, while studying code generation in Compiler Explorer for a C++ snippet that contains UB.
A git bisect pointed to commit 6a605b97a2006 

Repro on Compiler Explorer: https://godbolt.org/z/7MqbW4M18

```
// clang -g -O2 -mllvm -opt-bisect-limit=20 repro.cpp
// repro.cpp
class Class {};
void DoSomething(Class& c);
bool Cond();
void Func(Class* c) {
    if (!c && Cond())
        c = new Class();
    // UB if c == null and Cond() returns false
    DoSomething(*c);
}
```

Stack trace on a debug clang 14.0.0 build, under Windows:
```
 #0 0x00007ff77b6f0a90 HandleAbort F:\GitLab\llvm-project\llvm\lib\Support\Windows\Signals.inc:408:0
 #1 0x00007ff78376ff71 raise F:\GitLab\__llvmbuild\ninja-debug\minkernel\crts\ucrt\src\appcrt\misc\signal.cpp:547:0
 #2 0x00007ff783753bc9 abort F:\GitLab\__llvmbuild\ninja-debug\minkernel\crts\ucrt\src\appcrt\startup\abort.cpp:71:0
 #3 0x00007ff783755815 common_assert_to_stderr_direct F:\GitLab\__llvmbuild\ninja-debug\minkernel\crts\ucrt\src\appcrt\startup\assert.cpp:161:0
 #4 0x00007ff783755b87 common_assert_to_stderr<wchar_t> F:\GitLab\__llvmbuild\ninja-debug\minkernel\crts\ucrt\src\appcrt\startup\assert.cpp:179:0
 #5 0x00007ff783755941 common_assert<wchar_t> F:\GitLab\__llvmbuild\ninja-debug\minkernel\crts\ucrt\src\appcrt\startup\assert.cpp:415:0
 #6 0x00007ff78375556f _wassert F:\GitLab\__llvmbuild\ninja-debug\minkernel\crts\ucrt\src\appcrt\startup\assert.cpp:444:0
 #7 0x00007ff78077a53d `anonymous namespace'::LDVImpl::handleDebugInstr F:\GitLab\llvm-project\llvm\lib\CodeGen\LiveDebugVariables.cpp:888:0
 #8 0x00007ff78077a055 `anonymous namespace'::LDVImpl::collectDebugValues F:\GitLab\llvm-project\llvm\lib\CodeGen\LiveDebugVariables.cpp:942:0
 #9 0x00007ff78077784f `anonymous namespace'::LDVImpl::runOnMachineFunction F:\GitLab\llvm-project\llvm\lib\CodeGen\LiveDebugVariables.cpp:1270:0
#10 0x00007ff7807774cf llvm::LiveDebugVariables::runOnMachineFunction(class llvm::MachineFunction &) F:\GitLab\llvm-project\llvm\lib\CodeGen\LiveDebugVariables.cpp:1314:0
#11 0x00007ff77bc6a143 llvm::MachineFunctionPass::runOnFunction(class llvm::Function &) F:\GitLab\llvm-project\llvm\lib\CodeGen\MachineFunctionPass.cpp:72:0
#12 0x00007ff77aec75a5 llvm::FPPassManager::runOnFunction(class llvm::Function &) F:\GitLab\llvm-project\llvm\lib\IR\LegacyPassManager.cpp:1434:0
#13 0x00007ff77aecc86e llvm::FPPassManager::runOnModule(class llvm::Module &) F:\GitLab\llvm-project\llvm\lib\IR\LegacyPassManager.cpp:1480:0
#14 0x00007ff77aec7f7a `anonymous namespace'::MPPassManager::runOnModule F:\GitLab\llvm-project\llvm\lib\IR\LegacyPassManager.cpp:1549:0
#15 0x00007ff77aec7a43 llvm::legacy::PassManagerImpl::run(class llvm::Module &) F:\GitLab\llvm-project\llvm\lib\IR\LegacyPassManager.cpp:539:0
#16 0x00007ff77aeccb41 llvm::legacy::PassManager::run(class llvm::Module &) F:\GitLab\llvm-project\llvm\lib\IR\LegacyPassManager.cpp:1676:0
#17 0x00007ff77d1df38b `anonymous namespace'::EmitAssemblyHelper::RunCodegenPipeline F:\GitLab\llvm-project\clang\lib\CodeGen\BackendUtil.cpp:1536:0
#18 0x00007ff77d1d71eb `anonymous namespace'::EmitAssemblyHelper::EmitAssembly F:\GitLab\llvm-project\clang\lib\CodeGen\BackendUtil.cpp:1567:0
#19 0x00007ff77d1d5906 clang::EmitBackendOutput(class clang::DiagnosticsEngine &, class clang::HeaderSearchOptions const &, class clang::CodeGenOptions const &, class clang::TargetOptions const &, class clang::LangOptions const &, class llvm::StringRef, class llvm::Module *, enum clang::BackendAction, class std::unique_ptr<class llvm::raw_pwrite_stream, struct std::default_delete<class llvm::raw_pwrite_stream>>) F:\GitLab\llvm-project\clang\lib\CodeGen\BackendUtil.cpp:1732:0
#20 0x00007ff77d9abcd2 clang::BackendConsumer::HandleTranslationUnit(class clang::ASTContext &) F:\GitLab\llvm-project\clang\lib\CodeGen\CodeGenAction.cpp:374:0
#21 0x00007ff77ff0a76b clang::ParseAST(class clang::Sema &, bool, bool) F:\GitLab\llvm-project\clang\lib\Parse\ParseAST.cpp:178:0
#22 0x00007ff77d77c048 clang::ASTFrontendAction::ExecuteAction(void) F:\GitLab\llvm-project\clang\lib\Frontend\FrontendAction.cpp:1076:0
#23 0x00007ff77d9a6ad8 clang::CodeGenAction::ExecuteAction(void) F:\GitLab\llvm-project\clang\lib\CodeGen\CodeGenAction.cpp:1107:0
#24 0x00007ff77d77b9fd clang::FrontendAction::Execute(void) F:\GitLab\llvm-project\clang\lib\Frontend\FrontendAction.cpp:971:0
#25 0x00007ff77ba0839a clang::CompilerInstance::ExecuteAction(class clang::FrontendAction &) F:\GitLab\llvm-project\clang\lib\Frontend\CompilerInstance.cpp:1036:0
#26 0x00007ff77baf90a6 clang::ExecuteCompilerInvocation(class clang::CompilerInstance *) F:\GitLab\llvm-project\clang\lib\FrontendTool\ExecuteCompilerInvocation.cpp:263:0
#27 0x00007ff77aa92a91 cc1_main(class llvm::ArrayRef<char const *>, char const *, void *) F:\GitLab\llvm-project\clang\tools\driver\cc1_main.cpp:248:0
```

Note that for this repro, on LLVM 14, it crashes for `-opt-bisect-limit` values between 6 and 89 included.

Just in case the corresponding passes are relevant, here are the values:
`BISECT: running pass (6) SROAPass on _Z4FuncP5Class` 
`BISECT: running pass (89) CodeGen Prepare on function (_Z4FuncP5Class)`

I could also repro on trunk, the pass names matched.
I didn't manage to repro running `opt` tool on llvm-ir output by clang, as suggested in https://llvm.org/docs/HowToSubmitABug.html#crashing-bugs

It doesn't block our builds, as I only encountered that crash with `-opt-bisect-limit` and code containing UB, but it seemed worth reporting.

**Comment by jmorse** [comment]
This is caused by debug-info instruction referencing, it can be worked around with `-mllvm -experimental-debug-variable-locations=0`.

The cause seems to be that the machine function is created with optimisation-level `CodeGenOpt::None`, but when it hits register allocation it's become `CodeGenOpt::Default`. That makes various debug-info bits and pieces come from inconsistent modes.

It looks like the "OptLevelChanger" part of SelectionDAGISel.cpp does the re-setting.

**Comment by jmorse** [comment]
Patch in https://reviews.llvm.org/D123033

**Comment by fhahn** [comment]
IIUC this has been fixed by fb6596f1ecab652b5b90cf2e395d64112504c1f8/ https://reviews.llvm.org/rGfb6596f1ecab652b5b90cf2e395d64112504c1f8

@jmores/@sylvain-audi please re-open if there's anything left to do.

**Comment by jmorse** [comment]
Re-opening to request that this is cherry-picked into the release branch -- this is likely to trigger any time that someone uses opt-bisect-limit with the limit crossing instruction-selection, which while not common, can be vital for diagnosing compiler bugs. Fixing this on the release branch is going to ease matters for other developers.

@tstellar could you cherry pick fb6596f1ecab6 to the llvm 14 release branch -- it cherry picks cleanly, and on x86_64 ubuntu check-llvm completes cleanly. @OCHyams as you approved it, is this OK for the release branch?

**Comment by jmorse** [comment]
(Made my best guess and labels and release milestones, apologies if I missed something)

**Comment by tstellar** [comment]
@OCHyams Is this safe to backport? https://github.com/llvm/llvm-project/commit/fb6596f1ecab652b5b90cf2e395d64112504c1f8

**Comment by OCHyams** [comment]
Sorry, I completely missed this.

> Is this safe to backport?

I think so - LGTM.

**Comment by tstellar** [comment]
Merged: 0f56ce0fb207
